app/services/petition_service.rb

```diff
@@ -0,0 +1,18 @@
+class PetitionService
+
+  attr_reader :mobile_service
+
+  def initialize(mobile_service: MobileApiService.new)
+    @mobile_service = mobile_service
+  end
+
+  def fetch_petition_info(petition_id, fresh: false)
+    cache_key = "mobile_petition_info:#{petition_id}"
+
+    Rails.cache.delete cache_key if fresh
```

Comment:
You don't need to delete. Actually without deleting we avoid more race conditions.

You can just force the write. http://api.rubyonrails.org/classes/ActiveSupport/Cache/Store.html#method-i-fetch